src/enrollments/enrollments.service.ts

```diff
@@ -46,4 +48,21 @@ export class EnrollmentsService {
 
     return new EnrollmentResponseDto(enrollment.id, student.id, classroom.id);
   }
+
+  async removeStudent(userId: number, deleteEnrollDto: CreateEnrollmentDto): Promise<void> {
+    const { studentId, classroomId } = deleteEnrollDto;
+
+    const user = await this.userRepository.findOneOrFail({ id: userId });
+    const classroom = await this.classroomRepository.findOneOrFail({ id: classroomId });
+
+    if (user.teacher && user.teacher.id !== classroom.teacher.id) {
+      throw new UnauthorizedException("You are not authorized to remove this student!");
+    }
```

Comment:
I think abstracting classroom teacher and student validation to a separate method makes sense here. Please make a tech debt ticket to refactor this validation into a new method and replace wherever this check is necessary with that method.